Added SSH and Kubernetes resource (#838)
Twingate/terraform-provider-twingate: docs/resources/kubernetes_resource.md

<a id="nestedblock--access_group"></a>
### Nested Schema for `access_group`

Optional:

- `access_policy` (Block Set) Restrict access according to JIT access policy (see [below for nested schema](#nestedblock--access_group--access_policy))
- `group_id` (String) Group ID that will have permission to access the Resource.
- `security_policy_id` (String) The ID of a `twingate_security_policy` to use as the access policy for the group IDs in the access block. Default is 'Null' which points to `Default Policy` on Admin console.

<a id="nestedblock--access_group--access_policy"></a>
### Nested Schema for `access_group.access_policy`

Optional:

- `approval_mode` (String) This will set the approval model on the edge. The valid values are `AUTOMATIC` and `MANUAL`.
- `duration` (String) This will set the access duration on the edge. Duration must be between 1 hour and 365 days. The valid values are like `1h` and `2d`.
- `mode` (String) This will set the access_policy mode on the edge. The valid values are `MANUAL`, `AUTO_LOCK` and `ACCESS_REQUEST`.

<a id="nestedblock--access_policy"></a>
### Nested Schema for `access_policy`

Optional:

> - `approval_mode` (String) This will set the approval model on the edge. The valid values are `AUTOMATIC` and `MANUAL`.

Reviewer: ```suggestion
- `approval_mode` (String) This will set the approval model for the policy. The valid values are `AUTOMATIC` and `MANUAL`.
```

page_title: "twingate_kubernetes_resource Resource - terraform-provider-twingate"
subcategory: ""
description: |-
  Kubernetes Resources are Twingate resources accessed via a Kubernetes Gateway.
---

# twingate_kubernetes_resource (Resource)

Kubernetes Resources are Twingate resources accessed via a Kubernetes Gateway.

## Example Usage

```terraform
provider "twingate" {
  api_token = "1234567890abcdef"
  network   = "mynetwork"
}

resource "twingate_remote_network" "prod" {
  name = "Production Network"
}

resource "twingate_x509_certificate_authority" "tls" {
  name        = "My TLS CA"
  certificate = file("ca.pem")
}

resource "twingate_ssh_certificate_authority" "ssh" {
  name       = "My SSH CA"
  public_key = trimspace(file("~/.ssh/id_ed25519.pub"))
}

resource "twingate_gateway" "main" {
  remote_network_id = twingate_remote_network.prod.id
  address           = "10.0.0.1:8001"
  x509_ca_id        = twingate_x509_certificate_authority.tls.id
  ssh_ca_id         = twingate_ssh_certificate_authority.ssh.id
}

# Kubernetes resource accessed via in-cluster DNS
resource "twingate_kubernetes_resource" "prod_cluster" {
  name              = "Production K8s"

>   address           = "kubernetes.default.svc.cluster.local"

Reviewer 2: We should remove this

Author: `address` is a required field

Reviewer 2: @vmanilo could we remove this in the example now?
